### PR TITLE
feat: add interaction id into execute response(#1788)

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
1. Add interaction into the response of execute API.
2. Fix the issue that response won't be updated because it is not included in allowedList of `UpdateInteractionRequest`
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by SuZhou-Joe** [comment]
@ylwu-amzn would you mind reviewing this PR, it is a backport of #1788

**Review by Zhangxunmt — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Review by Zhangxunmt — DISMISSED** [review]
I am a little surprised that the response is allowed to be updated. We may revisit this in the app sec review for any risks.

**Comment by SuZhou-Joe** [comment]
> I am a little surprised that the response is allowed to be updated. We may revisit this in the app sec review for any risks.

We may need to add a flag to indicate if it is a internal call or from restful, if internal, it should be able to update any fields. Adding the `response` field into allowedList should be a temp solution.

**Comment by ylwu-amzn** [comment]
> I am a little surprised that the response is allowed to be updated. We may revisit this in the app sec review for any risks.

@Zhangxunmt , I remember we planned to control allowed fields for REST API. Seems the control is on transport action now.

**Comment by ylwu-amzn** [comment]
@SuZhou-Joe  Github CI failed

**Comment by SuZhou-Joe** [comment]
> @SuZhou-Joe Github CI failed

Checking that.

**Comment by xluo-aws** [comment]
> I am a little surprised that the response is allowed to be updated. We may revisit this in the app sec review for any risks.

The current logic is to create the a memory message first with response is empty, then add trace memory message and finally update the response of the memory message created in the first step.

**Comment by Zhangxunmt** [comment]
> > I am a little surprised that the response is allowed to be updated. We may revisit this in the app sec review for any risks.
> 
> The current logic is to create the a memory message first with response is empty, then add trace memory message and finally update the response of the memory message created in the first step.

Thanks, now I see. We should enable more strict allow list in the rest layer. The response shouldn’t be updated from apis.

**Comment by SuZhou-Joe** [comment]
@ylwu-amzn @Zhangxunmt CI fixed and please help to reapprove, thanks.

**Review by Zhangxunmt — APPROVED** [review]

File: memory/src/test/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionRequestTests.java [hunk 1]
```diff
@@ -9,8 +9,7 @@
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertNotSame;
 import static org.junit.Assert.assertSame;
-import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.INTERACTIONS_ADDITIONAL_INFO_FIELD;
-import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.INTERACTIONS_RESPONSE_FIELD;
+import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.*;
```
Comment (ylwu-amzn) [inline memory/src/test/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionRequestTests.java:12]: avoid import *

Comment (SuZhou-Joe) [reply memory/src/test/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionRequestTests.java]: Done.

**Review by ylwu-amzn — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

**Pull request merged**